tests/application_development/libcxx/src/main.cpp

```diff
@@ -35,12 +36,44 @@ static void test_vector(void)
 	zassert_equal(vector.size(), array.size(), "vector store failed");
 }
 
+struct make_unique_data {
+	static int ctors;
+	static int dtors;
+	int inst;
+
+	make_unique_data () :
+	inst{++ctors}
+	{ }
+
+	~make_unique_data ()
+	{
+		++dtors;
+	}
+};
+int make_unique_data::ctors;
+int make_unique_data::dtors;
+
+static void test_make_unique(void)
+{
+	zassert_equal(make_unique_data::ctors, 0, "ctor count not initialized");
+	zassert_equal(make_unique_data::dtors, 0, "dtor count not initialized");
+	auto d = std::make_unique<make_unique_data>();
+	zassert_true(static_cast<bool>(d), "allocation failed");
```

Comment:
Would it not be better to compare against `nullptr` instead of casting to a bool?

Comment also applies to release check.

Reply:
No, the intent was specifically to take advantage of the defined behavior for `operator bool()` on smart pointers (and other containers).   Bare `d` doesn't convert to a pointer because it's a smart pointer; you have to do `d.get()` which is not what real code would use when checking whether the pointer was valid.

However that doesn't work for argument passing because it's not one of the cases where contextual conversion occurs, so here an explicit cast is required.  It works for the release check because that code is a macro where the passed condition is `!d`, where the conversion does occur.

Reply:
Ok missed the `operator bool` in the c++ docs, note that `operator ==` is overloaded so comparing to a `nullptr` (or another unique pointer) is a valid operation.

Reply:
> `operator ==` is overloaded so comparing to a `nullptr` (or another unique pointer) is a valid operation.

Heh; and I missed that, since I don't think I'd ever use a smart pointer that way.  It'd happen in templates, though, so it's good it's there.